diesel_derives/src/lib.rs

```diff
@@ -394,11 +394,31 @@ pub fn derive_query_id(input: TokenStream) -> TokenStream {
 ///
 /// This trait can only be derived for structs, not enums.
 ///
-/// **When this trait is derived, it will assume that __all fields on your struct__
-/// matches __all fields in the query__, including the order. This means that field
-/// order is significant if you are using `#[derive(Queryable)]`. Field name has
-/// no effect. Think `Eq`, if you have `a, b, c` in your schema, you need to have
-/// `a, b, c` in the struct derived, not `a, b` or `a, c, b`.**
+/// **When this trait is derived, it will assume that __all fields on your
+/// struct__ is the same as __all fields in the query__, including the order.
+/// This means that field order is significant if you are using
+/// `#[derive(Queryable)]`. Field name has no effect. For example,
+///
+/// ```rust
+/// # extern crate diesel;
+/// # use diesel::prelude::*;
+/// table! {
+///     posts {
+///         id -> Integer,
+///         user_id -> Integer,
+///         title -> VarChar,
+///     }
+/// }
+///
+/// #[derive(Queryable, Debug)]
+/// pub struct Post {
+///     id: i32,
+///     user_id: i32,
+///     title: String,
+///     // this cannot be id, user_id (missing title)
+///     // this cannot be id, title, user_id (incorrect order)
+/// }
```

Comment:
I'm sorry to write that, but this whole example is missing a few really important points:

* Just having a `table!` and a struct deriving `Queryable` means nothing for diesel. There is no connection between those two data structures as long as you don't have a query loading the result for the table into the data structure.
* Field names do not matter as written in the block above. So just renaming some field to something different will be fine as long as the types match
* It is really important to point out it is totally fine to use a struct implementing/deriving `Queryable` with one or more queries, which could reference as lot tables as you want.

Reply:
> Just having a table! and a struct deriving Queryable means nothing for diesel. There is no connection between those two data structures as long as you don't have a query loading the result for the table into the data structure.

Okay, will amend it with `load`.

> It is really important to point out it is totally fine to use a struct implementing/deriving Queryable with one or more queries, which could reference as lot tables as you want.

Ah, I am not aware of this.

Reply:
This part is moved.